### ISSUE TITLE
Split user guide into a section guides

### ISSUE THREAD

**Issue by jlewi** [kubeflow/website#97]
Right now we have an uber [user_guide](https://www.kubeflow.org/docs/about/user_guide/) with different sections for TFJob, Seldon, Jupyter, etc....

This won't scale.

I think we should create a new section either "Guides" that would be at the same level as the About and Getting Started section. We could then have individual pages for things like

- TFJob
- PyTorch
- Seldon
- JupyterHub

Kubernetes has a section called "Tasks" but @sarahmaddox (Technical Writer who will be contributing to Kubeflow) suggested Guides might be better. Lets go with her recommendation.

@sarahmaddox will be evaluate overall layout of the docs at some point. However, I think splitting up the user guide will significantly improve user's ability to find relevant information. So I'd prefer not to wait on that.

**Comment by sarahmaddox** [comment]
+1 to the idea of a "Guides" section.

+1 to the idea of splitting the current user guide into smaller guides, each of which tackles a specific component. The content of the user guide is well suited to such a split. As well as the sections jlewi@ has mentioned, the user guide contains info on setting up a persistent disk, upgrading Kubeflow deployments, troubleshooting, ksonnet.

As a goal for the future, we can add some end-to-end tutorials that help people deploy Kubeflow and integrate specific components (installation on Kubernetes Engine or local, TensorFlow or Pytorch training, Seldon or TensorFlow Serving, and so on). We'll need to choose a few typical workflows and create the tutorials that match them.

**Comment by amsaha** [comment]
@jlewi  is **priority/1** a typo and was intended to be **priority/p1**?

**Comment by amsaha** [comment]
Does this look okay?

**Comment by sarahmaddox** [comment]
@amsaha Thanks, that's a great start towards addressing this issue. Your PR creates the "Guides" section and moves the PyTorch guide into the section, which is good.

This issue #97 also proposes that we split the "User Guide" doc (https://www.kubeflow.org/docs/about/user_guide/) into separate docs, and put those separate docs into the "Guides" section. That's a bigger task than is addressed by your current PR.

**Comment by amsaha** [comment]
@sarahmaddox Yes, I think the separation of user guides is a larger task and has to be taken up via a separate issue. Else the change might will be too big.

**Comment by amsaha** [comment]
Should we simply move the different sections of the current [user_guide](https://www.kubeflow.org/docs/about/user_guide/)? Maybe that is what was suggested. Sorry for the confusion. Let me do that.

**Comment by sarahmaddox** [comment]
That sounds good. @amsaha. Here's an idea of the guides we'll end up with:
* ksonnet - also include the section near the end of the user_guide: "Why Kubeflow uses ksonnet"
* Usage reporting
* Jupyter notebooks
* TensorFlow serving
* Seldon serving
* TensorFlow training
* PyTorch training
* Persistent disks
* Upgrading Kubeflow deployments
* Troubleshooting - For now, let's keep this all in one guide. (Note that some of the troubleshooting tips relate specifically to a component or environment e.g. Minikube / Jupyter Notebooks. But I think it makes sense to keep the troubleshooting tips together in one doc, especially where they relate to deployment of Kubeflow. If necessary, we can split them up later.)

**Comment by lluunn** [comment]
I will add a TF Serving section under Guildes

**Comment by sarahmaddox** [comment]
Thanks @lluunn - it's worth coordinating with @amsaha, as he's creating a TF serving section by pulling out the existing content from the user_guide.

**Comment by lluunn** [comment]
I will add new stuffs; but I can wait after @amsaha creates the section.

**Comment by jlewi** [comment]
LGTM

**Comment by sarahmaddox** [comment]
Thanks @amsaha!

**Comment by amsaha** [comment]
Have done most of the refactoring except the PyTorch part for which I have a doubt.
There are two copies of PyTorch instructions:

- The [PyTorch guide in the Getting Started section](https://www.kubeflow.org/docs/started/pytorch/)
- The other is [Submitting a PyTorch training job](https://www.kubeflow.org/docs/about/user_guide/#submitting-a-pytorch-training-job)

I believe the first one is what we need to keep and we can do away with the second one. Worst case we will have to collapse them into one.

@jlewi @johnugeorge, can one of you take a look and let me know?

**Comment by jlewi** [comment]
The former looks more comprehensive to me.  I think the second one is redundant.

**Comment by amsaha** [comment]
Thanks @jlewi . Will proceed with that.

**Comment by amsaha** [comment]
PR has been updated with latest changes.

**Comment by amsaha** [comment]
This can be closed. via https://github.com/kubeflow/website/pull/102.
